crates/astria-conductor/src/driver.rs

```diff
@@ -151,15 +161,22 @@ impl Driver {
     #[instrument(name = "driver", skip_all)]
     pub(crate) async fn run(mut self) -> Result<()> {
         use futures::StreamExt as _;
-        use sequencer_client::SequencerSubscriptionClientExt as _;
+        use sequencer_client::{
+            extension_trait::NewBlocksStream,
+            SequencerSubscriptionClientExt as _,
+        };
 
         info!("Starting driver event loop.");
-        let mut new_blocks = self
-            .sequencer_client
-            .client
-            .subscribe_new_block_data()
-            .await
-            .wrap_err("failed subscribing to sequencer to receive new blocks")?;
+        let mut new_blocks = if self.sequencer_client.is_some() {
+            let seq_client = self.sequencer_client.take().unwrap();
+            seq_client
+                .client
+                .subscribe_new_block_data()
+                .await
+                .wrap_err("failed subscribing to sequencer to receive new blocks")?
+        } else {
+            NewBlocksStream::empty()
```

Comment:
I don't think this will work - the docs https://docs.rs/futures/latest/futures/stream/fn.empty.html say "The returned stream will always return Ready(None) when polled." which means the `select!` statement below will always hit the 
```
warn!("sequencer new-block subscription closed unexpectedly; shutting down driver");
break;
```

case, shutting down the process.

can you change the `loop` below to only `select!` on `new_blocks` if reading from the sequencer client is enabled?

Reply:
After messing around a bit I decided to change the `NewBlockStream::empty()` definition to use `stream::pending()` instead of `stream::empty()`. This should give the desired functionality without needing to change or duplicate the `select!` loop.